PWGHF/HFL/TableProducer/electronSelectionWithTPCEMcal.cxx

```diff
@@ -0,0 +1,356 @@
+    "registry",
+    {{"hNevents", "No of events", {HistType::kTH1F, {{3, 1, 4}}}},
+     {"hZvertex", "z vertex", {HistType::kTH1F, {{100, -100, 100}}}},
+
+     {"hTrackInformation", "Sparse TPC info; dE/dx;n#sigma;#it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});#eta;#varphi;passEMcal;", hTrackAllInfoSpec},
+     {"hClusterInformationBefore", "Cluster Info before match; Energy (GeV);#eta;#varphi", hClusterInfoSpec},
+     {"hClusterInformationAfter", "Cluster Info after match; Energy (GeV);#eta;#varphi", hClusterInfoSpec},
+     {"hPIDafterMatch", "PID Info after match; E/P; dE/dx;n#sigma;#it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});#eta;#varphi;", hTrackInfoSpec},
+     {"hEPRatioafterPID", "E/P Ratio after PID Cuts apply only trackwodca filter", {HistType::kTH2F, {{60, 0, 3}, {100, 0, 10}}}},
+     {"hPIDafterPIDcuts", "PID Info after PID cuts; E/P; #it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});n_{#sigma}^{e};GeV;M02;M20", hPIDSpec},
+     {"hClsTrkEtaPhiDiffTimeEnergy", "ClsTrkEtaPhiDiffTimeEnergy;#Delta#eta;#Delta#varphi;Sec; Energy (GeV)", hDeltaPhiDeltaEtaClusterTrackSpecEnergy}}};
+
+  void init(InitContext&)
+  {
+    registry.get<THnSparse>(HIST("hTrackInformation"))->Sumw2();
+    registry.get<THnSparse>(HIST("hClusterInformationBefore"))->Sumw2();
+    registry.get<THnSparse>(HIST("hClusterInformationAfter"))->Sumw2();
+    registry.get<THnSparse>(HIST("hPIDafterMatch"))->Sumw2();
+    registry.get<THnSparse>(HIST("hPIDafterPIDcuts"))->Sumw2();
+    registry.get<THnSparse>(HIST("hClsTrkEtaPhiDiffTimeEnergy"))->Sumw2();
+  }
+
+  // Track Selection Cut
+  template <typename T>
+  bool selTracks(T const& track)
+  {
+    if (!track.isGlobalTrackWoDCA())
+      return false;
+    if (std::abs(track.dcaXY()) > dcaXYTrackMax || std::abs(track.dcaZ()) > dcaZTrackMax)
+      return false;
+    if (track.eta() < etaTrackMin || track.eta() > etaTrackMax)
+      return false;
+    if ((track.phi() < phiTrackEMCalMin || track.phi() > phiTrackEMCalMax) && (track.phi() < phiTrackDCalMin || track.phi() > phiTrackDCalMax))
+      return false;
+    if (track.pt() < ptTrackMin)
+      return false;
+    return true;
+  }
+
+  // MC Reco Track Selection Cut
+  template <typename T>
+  bool mcSelTracks(T const& mcTrack)
+  {
+    if (!mcTrack.isGlobalTrackWoDCA())
+      return false;
+    if (std::abs(mcTrack.dcaXY()) > mcRecDcaXYTrackMax || std::abs(mcTrack.dcaZ()) > mcRecDcaZTrackMax)
+      return false;
+    if (mcTrack.eta() < mcRecEtaTrackMin || mcTrack.eta() > mcRecEtaTrackMax)
+      return false;
+    if ((mcTrack.phi() < mcRecPhiTrackEMCalMin || mcTrack.phi() > mcRecPhiTrackEMCalMax) && (mcTrack.phi() < mcRecPhiTrackDCalMin || mcTrack.phi() > mcRecPhiTrackDCalMax))
+      return false;
+    if (mcTrack.pt() < mcRecPtTrackMin)
+      return false;
+    return true;
+  }
+
+  // Electron Identification
+  template <bool IsData, bool IsMC, typename TracksType, typename ClusterType, typename MatchType, typename CollisionType>
```

Comment:
```suggestion
  template <bool isData, bool isMc, typename TracksType, typename ClusterType, typename MatchType, typename CollisionType>
```
Why do you want to have flags `isData` and `isMc` independent?

Reply:
At this time, I am not sure if the MC cut is the same as the data cut.

Reply:
That's ok. My point is that you have two independent switches, which gives you 4 possible combinations, while you probably only need one switch for two options.

Reply:
Okay, thanks. I will use one switch for the two options.